proxy/tests/transparency.rs

```diff
@@ -360,5 +368,240 @@ fn http1_get_doesnt_add_transfer_encoding() {
         .inbound(srv)
         .run();
     let client = client::http1(proxy.inbound, "transparency.test.svc.cluster.local");
-    assert_eq!(client.get("/"), "hello h1");
+
+    let methods = &[
+        "GET",
+        "POST",
+        "PUT",
+        "DELETE",
+        "HEAD",
+        "PATCH",
+    ];
+
+    for &method in methods {
+        let resp = client.request(
+            client
+                .request_builder("/")
+                .method(method)
+        );
+
+        assert_eq!(resp.status(), StatusCode::OK, "method={:?}", method);
+    }
+}
+
+#[test]
+fn http1_content_length_zero_is_preserved() {
+    let _ = env_logger::try_init();
+
+    let srv = server::http1()
+        .route_fn("/", |req| {
+            let status = if req.headers()["content-length"] == "0" {
+                StatusCode::OK
+            } else {
+                StatusCode::BAD_REQUEST
+            };
+            Response::builder()
+                .status(status)
+                .header("content-length", "0")
+                .body(Default::default()) // default is hyper::Body::empty()
+                .unwrap()
+        })
+        .run();
+    let ctrl = controller::new().run();
+    let proxy = proxy::new()
+        .controller(ctrl)
+        .inbound(srv)
+        .run();
+    let client = client::http1(proxy.inbound, "transparency.test.svc.cluster.local");
+
+
+    let methods = &[
+        "GET",
+        "POST",
+        "PUT",
+        "DELETE",
+        "HEAD",
+        "PATCH",
+    ];
+
+    for &method in methods {
+        let resp = client.request(
+            client
+                .request_builder("/")
+                .method(method)
+                .header("content-length", "0")
+        );
+
+        assert_eq!(resp.status(), StatusCode::OK, "method={:?}", method);
+        assert_eq!(resp.headers()["content-length"], "0", "method={:?}", method);
+    }
+}
+
+#[test]
+fn http1_bodyless_responses() {
+    let _ = env_logger::try_init();
+
+    let req_status_header = "x-test-status-requested";
+
+    let srv = server::http1()
+        .route_fn("/", move |req| {
+            let status = req.headers()
+                .get(req_status_header)
+                .map(|val| {
+                    val.to_str()
+                        .expect("req_status_header should be ascii")
+                        .parse::<u16>()
+                        .expect("req_status_header should be numbers")
+                })
+                .unwrap_or(200);
+
+            Response::builder()
+                .status(status)
+                .body(Default::default()) // default is hyper::Body::empty()
```

Comment:
This is confusing. If there is no body then why not use `Body::empty()`, the default? If the default works, I think it would be clearer to use it.